lib/bundle/adder.rb

```diff
@@ -8,10 +8,19 @@ module Adder
     def add(*args, type:, global:, file:)
       brewfile = Brewfile.read(global:, file:)
       content = brewfile.input
-      # TODO: - validate formulae and casks
-      #       - support `:describe`
-      content << args.map { |arg| "#{type} \"#{arg}\"" }
-                     .join("\n") << "\n"
+      # TODO: - support `:describe`
+      new_content = args.map do |arg|
+        case type
+        when :brew
+          Formulary.factory(arg)
+        when :cask
+          Cask::CaskLoader.load(arg)
```

Comment:
Would be good to support `mas` etc. if not too heavy a lift?

Reply:
@carlocab Ah, gotcha, thanks! Might be worth noting that in a follow-up as a Ruby comment.

Reply:
Ah, whoops. I lied about `mas` 😅

It requires an `id` to be included in the entry. `whalebrew` and `vscode` should work fine though.

Let me think a bit more about how to handle `mas` better.

Reply:
@carlocab Yeh, `mas` may be tough; would be reasonable to just not support it IMO.